### ISSUE TITLE
How to add BCELoss + DiceLoss?

### ISSUE THREAD

**Issue by dfayzur** [qubvel/segmentation_models.pytorch#104]
Hi,
Thanks for the refactor of the code base. How can I use BCELoss + DiceLoss together? I saw SumOfLosses function, but I do not know the steps to add them in my model.

Help much appreciated

**Comment by qubvel** [comment]
Hi!
create instance of BCELoss and instance of DiceLoss and than use `total_loss = bce_loss + dice_loss`

**Comment by ysssgdhr** [comment]
> Hi!
> create instance of BCELoss and instance of DiceLoss and than use `total_loss = bce_loss + dice_loss`

Hello author! Your code is beautiful!
It's awesome to automatically detect the name of loss with regularization function!
Did I write loss the right way? I'm a noob with pytorch....

`class CompoundLoss(base.Loss):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.diceloss=smp.utils.losses.DiceLoss()
        self.bceloss = torch.nn.BCELoss()
        self.bceloss.__name__ = 'bce_loss'
    def forward(self, y_pr, y_gt):
        return 0.4 * self.diceloss.forward(y_pr, y_gt) + 0.6 * self.bceloss.forward(y_pr, y_gt)
loss = CompoundLoss()`

**Comment by Geeks-Sid** [comment]
Do you also find this to not work properly?

Right now I am trying to train, and the network trains fine, but during validation, the dice comes out to be zero.
